src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt

```diff
@@ -175,6 +183,33 @@ class SnykToolWindowPanel(
                 scanListener
             }
 
+        project.messageBus
+            .connect(this)
+            .subscribe(
+                SnykScanListenerLS.SNYK_SCAN_TOPIC,
+                object : SnykScanListenerLS {
+                    override fun onPublishDiagnostics(product: String, snykFile: SnykFile, issueList: List<ScanIssue>) {
+                        // Refresh the tree view on receiving new diags from the Language Server
+                        getSnykCachedResults(project)?.let {
+                            when (product) {
+                                PRODUCT_CODE -> it.currentSnykCodeResultsLS.clear()
```

Comment:
Why do we clear the complete cache when receiving a single file? Is that correct?